addons/TreasurePool/TreasurePool.lua

```diff
@@ -54,115 +54,113 @@ defaults.display.text.alpha = 255
-                local timer = {}    
-                timer[i] = os.date('!%M:%S', diff)
-                if timer[i] then
-                    if diff < 0 then -- stop the timer when 00:00 so it don't show 59:59 for a brief moment
-                        remove:add('index' .. i)
-                        remove:add('lotting' .. i)
-                    else
-                        info['index' .. i] = (
-                            diff < 60 and
-                                '\\cs(255,0,0)' .. res.items[treasure[i].item_id].name .. ' → ' .. timer[i]
-                            or diff > 180 and
-                                '\\cs(0,255,0)' .. res.items[treasure[i].item_id].name .. ' → ' .. timer[i]
-                            or
-                                '\\cs(255,128,0)' .. res.items[treasure[i].item_id].name .. ' → ' .. timer[i]) .. '\\cr'
-                    end
-                end
-            else -- show item name in case the addon is loaded with items on tresure box
-                info['index' .. i] = res.items[treasure[i].item_id].name
-            end
-            if lotter[i] and lot[i] then
-                if lotter[i] == ' ' then
-                    remove:add('lotting' .. i)
-                elseif lot[i] > 0 then
-                    info['lotting' .. i] = (
-                        '\\cs(0,255,255)' .. (' | ' .. lotter[i] .. ': ' .. lot[i])) .. '\\cr'
+    current_string = 'Treasure Pool:'
+    for key, value in pairs(items) do
+        if value and value.temp then
+            local diff = os.difftime(value.temp, os.time())
+            local timer = os.date('!%M:%S', diff)
+            if diff >= 0 then
+                current_string = current_string..'\n['..key..']'
+                current_string = (
+                    diff < 60 and
+                    current_string..'\\cs(255,0,0)['..value.index..'] '..value.name..' → '..timer
+                    or diff > 180 and
+                    current_string..'\\cs(0,255,0)['..value.index..'] '..value.name..' → '..timer
+                    or
+                    current_string..'\\cs(255,128,0)['..value.index..'] '..value.name..' → '..timer)..'\\cr'
+                if value.lotter and value.lot and value.lot > 0 then
+                    current_string = current_string..' | '
+                    current_string = (current_string..'\\cs(0,255,255)'..value.lotter..': '..value.lot)..'\\cr'
                 end
+            else
+                table.remove(items, key)
             end
-            treasure_text:show()
-        else
-            remove:add('index' .. i)
-            remove:add('lotting' .. i)
+        box:show()
         end
-    treasure_text:update(info)
     end
-    for entry in remove:it() do
-        treasure_text[entry] = nil
-    end
-end)
+    box.current_string = current_string
+end
+
+Update:loop(0.5)
+
+windower.register_event('logout','zone change', function()
+    items = T{}
+end)
```

Comment:
Why no new line? :(

Reply:
Dunno. Its new line at end a good coding practice? or for another reason?

Reply:
Depends on whom you ask, to me it is. That's why we added it to our coding guidelines for v5 :) But it was just a random comment, as I said before, the code is fine as is.

Reply:
Not having a newline at the end of a text file can cause problems with some tools. These tools and systems treat the newline character as a line terminator, not a line separator. Many editors/viewers (including GitHub) will give warnings about it for this reason.